pilot/pkg/networking/v1alpha3/listener.go

```diff
@@ -45,3 +81,456 @@ func BuildListeners(env model.Environment, node model.Proxy) ([]*xdsapi.Listener
+//
+//	// assumes that inbound connections/requests are sent to the endpoint address
+//	for _, mPort := range managementPorts {
+//		switch mPort.Protocol {
+//		case model.ProtocolHTTP, model.ProtocolHTTP2, model.ProtocolGRPC, model.ProtocolTCP,
+//			model.ProtocolHTTPS, model.ProtocolMongo, model.ProtocolRedis:
+//			cluster := v1.BuildInboundCluster(mPort.Port, mPort.Protocol, mesh.ConnectTimeout)
+//			listener := buildTCPListener(&v1.TCPRouteConfig{
+//				Routes: []*v1.TCPRoute{v1.BuildTCPRoute(cluster, []string{managementIP})},
+//			}, managementIP, uint32(mPort.Port), model.ProtocolTCP)
+//
+//			listeners = append(listeners, listener)
+//		default:
+//			log.Warnf("Unsupported inbound protocol %v for management port %#v",
+//				mPort.Protocol, mPort)
+//		}
+//	}
+//
+//	return listeners
+//}
+
+// TODO: move to plugins
+// applyInboundAuth adds ssl_context to the listener if the policy requires one.
+func applyInboundAuth(authenticationPolicy *authn.Policy, listener *xdsapi.Listener) {
+
+	if model.RequireTLS(authenticationPolicy) {
+		// TODO(mostrowski): figure out SSL
+		log.Debugf("TODO Apply authN policy %#v for %#v\n", authenticationPolicy, listener)
+	}
+}
+
+// options required to build an HTTPListener
+type buildHTTPListenerOpts struct { // nolint: maligned
+	// config           model.Config
+	// env              model.Environment
+	mesh             *meshconfig.MeshConfig
+	proxy            model.Proxy
+	proxyInstances   []*model.ServiceInstance
+	routeConfig      *xdsapi.RouteConfiguration
+	rdsConfig        *http_conn.HttpConnectionManager_Rds
+	ip               string
+	port             int
+	rds              string
+	useRemoteAddress bool
+	direction        http_conn.HttpConnectionManager_Tracing_OperationName
+	store            model.IstioConfigStore
+	authnPolicy      *authn.Policy
+}
+
+func buildHTTPListener(opts buildHTTPListenerOpts) *xdsapi.Listener {
+	filters := []*http_conn.HttpFilter{}
+	filters = append(filters, &http_conn.HttpFilter{
+		Name: util.CORS,
+	})
+	// TODO: need alphav3 fault filters.
+	// filters = append(filters, buildFaultFilters(opts.config, opts.env, opts.proxy)...)
+	filters = append(filters, &http_conn.HttpFilter{
+		Name: util.Router,
+	})
+
+	/*	TODO(mostrowski): need to port internal build functions for mixer.
+		if opts.mesh.MixerCheckServer != "" || opts.mesh.MixerReportServer != "" {
+			mixerConfig := v1.BuildHTTPMixerFilterConfig(opts.mesh, opts.proxy, opts.proxyInstances, opts.outboundListener, opts.store)
+		filter := &http_conn.HttpFilter{
+			Name: v1.MixerFilter,
+			Config:messageToStruct(mixerConfig),
+		}
+			filters = append([]*http_conn.HttpFilter{filter}, filters...)
+		}
+	*/
+	//refresh := time.Duration(opts.mesh.RdsRefreshDelay.Seconds) * time.Second
+
+	if filter := buildJwtFilter(opts.authnPolicy); filter != nil {
+		filters = append([]*http_conn.HttpFilter{filter}, filters...)
+	}
+
+	var rds *http_conn.HttpConnectionManager_Rds
```

Comment:
this function is getting too long. suggest keeping buildHTTPConnectionManager as a separate function.